thirdeye/thirdeye-frontend/app/pods/components/rootcause-settings/component.js

```diff
@@ -19,6 +36,97 @@ export default Ember.Component.extend({
 
   analysisRangeEnd: null,
 
+  // datepicker range
+  datePickerAnomalyRangeStart: Ember.computed.reads('anomalyRangeStart'),
+  datePickerAnomalyRangeEnd: Ember.computed.reads('anomalyRangeEnd'),
+  datePickerAnalysisRangeStart: Ember.computed.reads('analysisRangeStart'),
+  datePickerAnalysisRangeEnd: Ember.computed.reads('analysisRangeEnd'),
+
+  /**
+   * Selected Granularity
+   * @type {String}
+   */
+  granularity: null,
+
+  /**
+   * Granularities Options
+   * @type {String[]}
+   */
+  granularities: ['MINUTES', 'HOURS', 'DAYS'],
+
+  /**
+   * Selected Compare Mode
+   * @type {String}
+   */
+  compareMode: 'WoW',
+  /**
+   * Compare Mode Options
+   * @type {String[]}
+   */
+  compareModeOptions: ['WoW', 'Wo2W', 'Wo3W', 'Wo4W'],
+
+  /**
+   * Predefined Custom Ranges for
+   * the display region
+   * @type {Object}
+   */
+  predefinedAnalysisRanges: {
+    'Last 2 days': [
+      moment().subtract(3, 'days').startOf('day'), 
+      moment().subtract(1, 'days').endOf('day')
+    ],
+    'last 7 Dgit sattuays': [
+      moment().subtract(7, 'days').startOf('day'),
+      moment().subtract(1, 'days').endOf('day')
+    ]
+  },
+
+  /**
+   * filter options
+   * @type {Object}
+   */
+  metricFilters: {},
+
+  /**
+   * Selected filters
+   * @type {String} - a JSON string
+   */
+  filters: JSON.stringify({}),
+
+  /**
+   * Indicates the date format to be used based on granularity
+   * @type {String}
+   */
+  uiDateFormat: Ember.computed('granularity', function () {
+    const granularity = this.get('granularity');
+
+    switch (granularity) {
+      case 'DAYS':
+        return 'MMM D, YYYY';
+      case 'HOURS':
+        return 'MMM D, YYYY h a';
+      default:
+        return 'MMM D, YYYY hh:mm a';
+    }
+  }),
+
+  /**
+   * Indicates the allowed date range picker increment based on granularity
+   * @type {Number}
+   */
+  timePickerIncrement: Ember.computed('granularity', function () {
+    const granularity = this.get('granularity');
+
+    switch (granularity) {
+      case 'DAYS':
+        return 1440;
+      case 'HOURS':
+        return 60;
+      default:
+        return 5;
+    }
+  }),
+
   _updateFromContext() {
```

Comment:
do we have any convention regarding public/private method naming?

Reply:
imo this is fine. unless you want to use `typescript` which allows declaring `private` methods